polygon/sync/service.go

```diff
@@ -55,10 +58,13 @@ func NewService(
 		tmpDir,
 		logger,
 	)
+
+	polygonBridge := bridge.NewBridge(dataDir, logger, borConfig, heimdallClient.FetchStateSyncEvents, bor.GenesisContractStateReceiverABI())
 	blockDownloader := NewBlockDownloader(
 		logger,
 		p2pService,
 		heimdallService,
+		polygonBridge,
```

Comment:
I think we discussed this a few weeks ago with Daniel but maybe it got missed in the messages - we were discussing that ideally the block downloader does not have a dependency on the bridge. For the time being this can conceptually be better placed as a dependency to the `polygon/sync Store` - in its implementation of `InsertBlocks` we can notify the bridge about the new blocks

Reply:
Done, however this does mean that we have two usages of ProcessNewBlocks (one in `polygon.sync` and the other in `polygon.sync.stage`) instead of the single one with the previous approach.

Reply:
yes that is intended in order to make the stage integration work